From c619be2fadae1b5a87151db0b0adddf55584084a Mon Sep 17 00:00:00 2001
From: Murray Cumming <murrayc@murrayc.com>
Date: Mon, 14 Oct 2013 08:11:25 +0000
Subject: Gio::ActionGroup: Remove a nonsensical check.

* gio/src/actiongroup.hg: get_action_state_hint():
  Remove a wrongly copy/pasted g_return_if_fail() line that
  causes a compiler error when the template is actually used.
  Ideally we would have tests for uses of templated methods.
  Bug #709963 (Johannes Meng, David Evans)
---
diff --git a/gio/src/actiongroup.hg b/gio/src/actiongroup.hg
index 4d706d7..434b7fa 100644
--- gio/src/actiongroup.hg.orig
+++ gio/src/actiongroup.hg
@@ -184,9 +184,6 @@ void ActionGroup::get_action_state_hint(const Glib::ustring& action_name, T_Valu
 
   typedef Glib::Variant<T_Value> type_glib_variant;
 
-  g_return_if_fail(
-    g_variant_type_equal(g_action_get_state_type(const_cast<GAction*>(gobj())), type_glib_variant::variant_type().gobj()));
-
   const Glib::VariantBase variantBase = get_action_state_hint_variant(action_name);
 
   // We can't check the type (a range) that will be returned before getting the range hint.
--
cgit v0.9.2
